driver/bpf/fillers.h

```diff
@@ -4346,6 +4483,99 @@ FILLER(sys_sendmsg_x, true) {
 	return res;
 }
 
+FILLER(sys_sendmmsg_x, true) {
+	struct sockaddr *usrsockaddr;
+	struct mmsghdr mmh = {0};
+	struct mmsghdr *mmh_ptr;
+	unsigned long vlen;
+	unsigned long val;
+	uint16_t size = 0;
+	int addrlen;
+	int err = 0;
+	int res;
+	long retval = bpf_syscall_get_retval(data->ctx);
+	int fd;
+
+	if(retval < 0) {
+		/* Parameter 1: res (type: PT_ERRNO) */
+		res = bpf_push_s64_to_ring(data, retval);
+		CHECK_RES(res);
+
+		/* Parameter 2: fd (type: PT_BYTEBUF) */
+		bpf_push_empty_param(data);
+		CHECK_RES(res);
+
+		/* Parameter 3: size (type: PT_UINT32) */
+		bpf_push_u32_to_ring(data, 0);
+		CHECK_RES(res);
+
+		/* Parameter 4: data (type: PT_BYTEBUF) */
+		bpf_push_empty_param(data);
+		CHECK_RES(res);
+
+		/* Parameter 5: tuple (type: PT_SOCKTUPLE) */
+		return bpf_push_empty_param(data);
+	}
+
+	fd = bpf_syscall_get_argument(data, 0);
+	mmh_ptr = (struct mmsghdr *)bpf_syscall_get_argument(data, 1);
+	vlen = bpf_syscall_get_argument(data, 2);
+
+	/* Retrieve the message header */
+	if(bpf_probe_read_user(&mmh, sizeof(mmh), (void *)(mmh_ptr))) {
```

Comment:
so if `mmh_ptr` we rely on the fact that `mmh` is zeroed and we won't enter all the following branches, is it true?

Reply:
I missed this question the first time I went through the review, sorry.

`mmh_ptr` is read from the arguments, as such, I assume it will always be pointing to a valid position in memory because the syscall was successful here and we haven't returned control to userspace, so the memory must still be there. If, for whatever reason, the pointer was invalid or it was no longer readable due to a cache miss or such a situation, I would expect `bpf_probe_read_user` to fail and the event to be dropped altogether, (the if condition will be true).

Am I missing something here?

Reply:
makes sense